packages/angular_devkit/architect/src/api_spec.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Target, targetFromTargetString, targetStringFromTarget } from './api';
+
+describe('Architect API', () => {
+  const goldens: { [t: string]: Target } = {
+    'a:b:c': { project: 'a', target: 'b', configuration: 'c' },
+    'a:b': { project: 'a', target: 'b' },
+    ':b': { project: '', target: 'b' },
+    ':b:c': { project: '', target: 'b', configuration: 'c' },
+    ':': { project: '', target: '' },
+    '::': { project: '', target: '', configuration: '' },
+    'a:b:': { project: 'a', target: 'b', configuration: '' },
+    'a::c': { project: 'a', target: '', configuration: 'c' },
```

Comment:
Is having an empty string for a `target` actually valid?

Reply:
Technically yes. This would be the only place we check for that.